recipes/nuraft/all/conanfile.py

```diff
@@ -0,0 +1,90 @@
+from os.path import join
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, get, apply_conandata_patches
+from conans import CMake
+from conans.tools import check_min_cppstd
+
+required_conan_version = ">=1.50.0"
+
+class NuRaftConan(ConanFile):
+    name = "nuraft"
+    homepage = "https://github.corp.ebay.com/sds/NuRaft"
+    description = """Cornerstone based RAFT library."""
+    topics = ("raft")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+
+    settings = "os", "compiler", "arch", "build_type"
+
+    options = {
+        "shared": ['True', 'False'],
+        "fPIC": ['True', 'False'],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(p["patch_file"])
+
+    def requirements(self):
+        self.requires("boost/1.79.0")
+        self.requires("openssl/1.1.1q")
+
+    def validate(self):
+        if self.info.settings.os in ["Windows"]:
+            raise ConanInvalidConfiguration("{} Builds are unsupported".format(self.info.settings.os))
+        if self.info.settings.os in ["Macos"] and self.options.shared:
+            raise ConanInvalidConfiguration("Building Shared Object for {} unsupported".format(self.info.settings.os))
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def cmakeGet(self):
```

Comment:
See our https://github.com/conan-io/conan-center-index/blob/master/docs/reviewing.md#cmake

```suggestion
    @functools.lru_cache(1)
    def _configure_cmake(self):
```